authentik/enterprise/policies/unique_password/tasks.py

```diff
@@ -0,0 +1,98 @@
+from structlog import get_logger
+
+from authentik.enterprise.policies.unique_password.models import (
+    UniquePasswordPolicy,
+    UserPasswordHistory,
+)
+from authentik.events.system_tasks import SystemTask, TaskStatus, prefill_task
+from authentik.policies.models import PolicyBinding
+from authentik.root.celery import CELERY_APP
+
+LOGGER = get_logger()
+
+
+@CELERY_APP.task(bind=True, base=SystemTask)
+@prefill_task
+def purge_password_history_table(self: SystemTask):
+    """Remove all entries from the UserPasswordHistory table"""
+    unique_pwd_policy_count = UniquePasswordPolicy.objects.count()
+
+    if unique_pwd_policy_count > 0:
+        # No-op; A UniquePasswordPolicy exists, no need to purge the table
+        self.set_status(
+            TaskStatus.SUCCESSFUL,
+            """No need to purge UserPasswordHistory table.
+            A Unique Password Policy instance still exists.""",
+        )
+        return
+
+    UserPasswordHistory.objects.all().delete()
+    self.set_status(TaskStatus.SUCCESSFUL, "Successfully purged UserPasswordHistory")
+
+
+@CELERY_APP.task()
+def trim_user_password_history(user_pk: int):
```

Comment:
As this is only called from `trim_all_password_histories`, I think we can just have one task deleting for all users

Reply:
And it can also call `check_and_purge_password_history` as well.

I fail to see the difference between `purge_password_history_table` and `check_and_purge_password_history` though

Reply:
You're right, I refactored  check_and_purge_password_history and purge_password_history_table into one function 👍

Reply:
as for `trim_all_password_histories` this approach allows us to have better error isolation(like if there's a problem with a particular user) and would scale better in huge user base

Reply:
In other tasks we don't separate like this. I still want both of those tasks to be merged. If there are several thousands of users with password histories, I don't want several thousands of tasks to be scheduled.